_posts/en/newsletters/2019-02-05-newsletter.md

```diff
@@ -0,0 +1,283 @@
+  Ultimately the situation was resolved, but it was a mess that many
+  would prefer not to repeat.
+
+    The speaker identifies the root cause of the problem as miners being
+    able to delay activation at no cost to themselves.  He proposes a
+    solution: use the randomness remaining in a block header hash to
+    determine whether or not a block signals for activation.  For
+    example, we'd choose a target that only 1 header hash out of 26,000
+    would match.  A block matching that target would be produced once every
+    six months on average, although nobody would know exactly when
+    (about 10% of the time, it'd be within 3 weeks; but another 10% of
+    the time, it'd take more than a year).
+
+    Miners would have no control over whether or not their block
+    signaled for activation, but they would have control over whether
+    they published that block.  A miner who refused to publish his own
+    block if it signaled for activation would lose the income from that
+    block but would successfully prevent the fork from activating at
+    least until the next signaling block was produced (which could be
+    tomorrow or could be two years later).  This gives miners a real
+    chance to hold back a change but only by sacrificing real income.
+    The end of the talk suggests some variations on the method with
+    different tradeoffs.
+
+    The idea needs to be analyzed for possible problems, but it does
+    provide an interesting alternative to [BIP9][]-style miner-activated
+    soft forks (MASFs) and [BIP8][]-style user-activated soft forks (UASFs).
+
+    At the conclusion of his talk, this speaker also announced that the
+    next Scaling Bitcoin conference and EdgeDev++ training sessions will
+    be later in 2019 in Tel Aviv, Israel.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+and [libsecp256k1][libsecp256k1 repo].*
+
+- [Bitcoin Core #14929][] allows peers that your node automatically
+  disconnected for misbehavior (e.g. sending invalid data) to reconnect
+  to your node if you have unused incoming connection slots.  If your
+  slots fill up, the misbehaving nodes are disconnected to make room for
+  nodes without a history of problems (unless the misbehaving node helps
+  your node in some other way, such as by connecting to a part of the
+  Internet from which you don't have many other peers).  Previously,
+  Bitcoin Core banned the IP addresses of misbehaving peers for a period
+  of time (default of 1 day); this was easily circumvented by attackers
+  with multiple IP addresses.  This solution gives those peers a chance
+  to be useful but provides priority to potentially more helpful peers.
+  If you manually ban a peer, such as by using the `setban` RPC,
+  connections from that peer will still be rejected.
+
+- [Bitcoin Core #13926][] adds a new `bitcoin-wallet` tool to the
```

Comment:
Nit: note that this is an offline tool: "adds a new offline `bitcoin-wallet` tool..."

Reply:
I found "offline" a bit ambiguous here (the way most people use `bitcoin-cli` is offline too, in the sense that it's only querying over localhost).  I changed a few words in the following sentence that should hopefully convey the same point, but please feel free to edit more if you think it's not clear enough.

Reply:
Looks great. Thanks!